Docs/BazelForiOSDevelopers.md

```diff
@@ -0,0 +1,522 @@
+
+The next line of code calls `git_repository` and defines the repository
+`build_bazel_rules_apple` from the git repository,
+`https://github.com/bazelbuild/rules_apple` for a given `commit`. 
+
+### Adding an iOS application with BUILD files
+
+`BUILD` files are where all the targets are defined. For Apple developers, this
+often includes apps, tests, app extensions, frameworks, and libraries.
+
+First, let's walk through creating a basic iOS application. In the root of the
+project, let's create the `BUILD` file. First, a library for the iOS
+application sources. The following code defines an `objc_library`, `sources`.
+
+```
+# /path/to/myproject/BUILD
+objc_library(
+    name="sources",
+    srcs=["main.m"]
+)
+```
+
+Next, create the application target with `rules_apple`'s `ios_application` rule.
+```
+# /path/to/myproject/BUILD
+...
+load("@build_bazel_rules_apple//apple:ios.bzl", "ios_application")
+ios_application(
+    name = "ios-app",
+    bundle_id = "com.bazel-bootcamp.some",
+    families = ["iphone"],
+    infoplists = ["Info.plist"],
+    minimum_os_version = "[MINIMUM VERSION]",
+    # Add `sources` as a dependency
+    deps = [ ":sources" ],
+)
+```
+
+In Xcode this is similar to navigating in the GUI and hitting `File -> New
+Target`. Together, the `ios_application` and `objc_library` targets together
+would be represented in Xcode as:
+
+![Docs](XcodeExampleOfiOSProject.png)
+	
+### Bazel command line
+
+In Xcode, command line builds are achieved through the command line interface
+`xcodebuild`.  to build the scheme `ios-app` from `MyProject.xcworkspace`,
+`xcodebuild`, would be invoked as:
+
+```
+xcodebuild -workspace MyProject.xcworkspace -scheme ios-app
+```
+
+Bazel exposes a command line which can also do builds. In Bazel, _every_ target
+is in the global WORKSPACE and there is no notion of schemes.
+
+```
+bazel build ios-app
+```
+
+The Bazel command line has hundreds of options, which can be found in the
+[Bazel
+documentation](https://docs.bazel.build/versions/master/command-line-reference.html).
+For iOS developers, a set of useful flags is available at
+[bazel-ios-users](https://github.com/ios-bazel-users/ios-bazel-users/blob/master/UsefulFlags.md).
+
+### Configurable build attributes
+
+In addition to the default Bazel options, it's common to create custom
+configuration settings to customize builds. Together, `select` and
+`config_setting` yield configurable build attributes.
+
+For example, the following build file has conditional `copts` on `:app_store`
+
+```
+objc_library(
+   name = "some",
+   srcs = ["some.m"],
+   copts = select({
+      # For app_store we need to build `some.m` with a special copt
+      ":app_store": ["-DAPPSTORE=1"],
+      "//conditions:default": [],
+    })
+)
+config_setting(
+    name = "app_store",
+    values = { "app_store" : "true" }
+)
+```
+
+This is can be set as a define and passed to Bazel on the command line
+
+```
+bazel build ios-app --define app_store=true
```

Comment:
It'd be useful to clarify the logic path that happens here. For example, you first add a config_setting that declares a config that's associated with a value, "app_store". In order to trigger the config ":app_store", the define "app_store=true" needs to be passed into Bazel, and an existing library needs to also be calling "select()" with the config setting named ":app_store". Is this following the logic correctly?

Reply:
This segment is mean to be simple example and explanation - I think the bazel docs did a good job at explaining `config_setting` in depth -  if not we should PR an improvement to the linked page:
> Configurable attributes are evaluated after the processing of macros and before the processing of rules (technically, between the loading and analysis phases. Any processing that Bazel does before the select() is evaluated will not know which branch will be chosen. In particular, macros can't change their behavior based on the chosen branch, and bazel query can only make conservative guesses about the configurable dependencies of a target. Conversely, when authoring a new type of rule, you do not need to worry about the ambiguity of configurable attributes because all select() expressions have already been replaced by their resolved values. See this FAQ for more on using select() with rules and macros.